ui/src/app/utils/workspace-utils.tsx

```diff
@@ -1,16 +1,31 @@
-import { FeaturedWorkspaceCategory, Workspace } from 'generated/fetch';
+import {
+  BillingStatus,
+  FeaturedWorkspaceCategory,
+  Workspace,
+} from 'generated/fetch';
 
 import { serverConfigStore } from 'app/utils/stores';
 
-export const isUsingFreeTierBillingAccount = (
-  workspace: Workspace
-): boolean => {
+export const isUsingInitialCredits = (workspace: Workspace): boolean => {
   return (
     workspace.billingAccountName ===
     'billingAccounts/' + serverConfigStore.get().config.freeTierBillingAccountId
   );
 };
 
+export const isValidBilling = (workspace: Workspace): boolean => {
```

Comment:
Added this utility function to replace direct comparison of billing status. This also involves looking at expired, exhausted, and bypass.